feat: Add Outlook email connector with OAuth2 authentication Wirh complete CRUD

MJunaidWali opened the pull request:
- Add comprehensive Outlook OAuth2 integration with Microsoft Graph API
- Implement email operations: send, read, reply, forward, delete
- Add folder management and file attachment support
- Include robust error handling and logging
- Add configuration for Outlook OAuth2 client settings
- Integrate Outlook connector in main API router and connectors registry
- Add outlook_tools.py for graph-based email operations

Features:
- OAuth2 authentication flow with Microsoft Graph API
- Email CRUD operations (create, read, update, delete)
- Folder management (create, list, move emails)
- File attachment handling (upload, download)
- Advanced search and filtering capabilities
- Token refresh and error handling
- Structured logging for debugging and monitoring

MJunaidWali committed:
feat: Add Outlook email connector with OAuth2 authentication

- Add comprehensive Outlook OAuth2 integration with Microsoft Graph API
- Implement email operations: send, read, reply, forward, delete
- Add folder management and file attachment support
- Include robust error handling and logging
- Add configuration for Outlook OAuth2 client settings
- Integrate Outlook connector in main API router and connectors registry
- Add outlook_tools.py for graph-based email operations

Features:
- OAuth2 authentication flow with Microsoft Graph API
- Email CRUD operations (create, read, update, delete)
- Folder management (create, list, move emails)
- File attachment handling (upload, download)
- Advanced search and filtering capabilities
- Token refresh and error handling
- Structured logging for debugging and monitoring

MKRNaqeebi commented on app/api/routes/outlook/outlook.py:
no need for this file

MKRNaqeebi replied:
we are not using these end points in frontend

MJunaidWali replied:
file deleted.

MKRNaqeebi commented on app/core/config.py:
```diff
@@ -163,6 +163,12 @@ def _enforce_non_default_secrets(self) -> Self:
+    OUTLOOK_CLIENT_SECRET: str | None = os.getenv("OUTLOOK_CLIENT_SECRET")
+    OUTLOOK_REDIRECT_URI: str = os.getenv("OUTLOOK_REDIRECT_URI", "http://localhost:8000/api/v1/outlook/callback/")
+    OUTLOOK_AUTH_URI: str = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize"
+
```
Add outlook scope here as well

MJunaidWali replied:
added scope.

MKRNaqeebi commented on app/api/routes/outlook/__init__.py:
remove all try excepts from file as these will hide actual errors

MJunaidWali replied:
removed all try excepts.

MJunaidWali commented:
Ok .i am going to start  work on them INSHALLAH they are done by tommorow.

MJunaidWali committed:
feat(outlook): Refactor and fix Outlook connector

- Refactors the Outlook OAuth2 flow to be manual and stateless, mirroring the GitHub connector.
- Implements the missing  functionality in the Outlook connector.
- Removes unused placeholder tools from the Outlook tools file.
- Fixes a bug in the main connectors route that prevented the UI from updating after a successful connection.
- Standardizes the connector name to 'outlook' to resolve inconsistencies.
